exonum/src/api/node/public/explorer.rs

```diff
@@ -124,11 +133,19 @@ impl ExplorerApi {
             (explorer.height(), explorer.blocks(..))
         };
 
+        let mut times = Vec::new();
+
         let blocks: Vec<_> = blocks_iter
             .rev()
             .filter(|block| !query.skip_empty_blocks || !block.is_empty())
             .take(query.count)
-            .map(|block| block.into_header())
+            .map(|block| {
+                if query.add_blocks_time {
+                    times.push(median_precommits_time(&block.precommits()));
+                }
```

Comment:
(Code style nit. Feel free to disregard.)

I'd use `inspect` for populating `times` like this:

```rust
.inspect(|block| {
    if query.add_blocks_time {
        times.push(median_precommits_time(&block.precommits()));
    }
})
.map(|block| block.into_header())
```

IMO, this makes usage of iterator combinators more focused: `map` is used only for element conversion and `inspect` is used for side-effects.

Reply:
I love such "style nits"! :+1: